src/Microsoft.NetCore.Analyzers/UnitTests/Runtime/DisposeObjectsBeforeLosingScopeTests.cs

```diff
@@ -7445,5 +7523,1012 @@ void M()
 }
 ");
         }
+
+        [Fact]
+        public void DifferentDisposePatternsInFinally_NoDiagnostic()
```

Comment:
FYI: We already have unit tests to verify disposable creations wrapped in a using statement are not flagged. https://github.com/dotnet/roslyn-analyzers/issues/2152 tracks verifying no violations for using declarations. This unit test verifies that we do not flag recommended and commonly used try-finally patterns for disposable allocations. The unit test after this one shows the kind of try-finally patterns that we do flag. We also have existing unit tests to show the maybe violations are now flagged in code without try-finally when not disposed on some paths, for example see https://github.com/dotnet/roslyn-analyzers/pull/2151/files#diff-834306aff1bd9b376d2e6830e9cbb560R6766